Automattic/woocommerce-payments
# Removed an unnecessary `$order->save()`, called with outdated data.

Fixes #4553

#### Changes proposed in this Pull Request

Move an unnecessary `$order->save()` call in `mark_payment_dispute_closed` to whenever it is needed for two reasons:
1. There are no changes to the order, which need to be saved whenever a refund was created (part of losing disputes), as those are already done within `wc_create_refund()`.
2. The `$order` object contains outdated data after `wc_create_refund()`, which loaded and modified another instance of the order. Before this change, the order was initially transitioned to `refunded`, but then back to `processing` because of the outdated data in the loaded instance.

#### Testing instructions

Exactly the same as in https://github.com/Automattic/woocommerce-payments/issues/4553.

- Without this branch you observe the described behavior
- With this branch, you should observe the expected behavior.

**Dev notes:**

Add breakpoints to:
- `includes/class-wc-payments-order-service.php` at the end of `mark_payment_dispute_closed`. While stopped, open the order page, and observe how the status has already transitioned to refunded.
- This has already happened within core's `wp-content/plugins/woocommerce/includes/wc-order-functions.php:640`

There are no additional tests here, as this flow is not covered.

-------------------


**Post merge**
